### PR TITLE
clipboard improvements

### PR DESCRIPTION
**Pull request by jawadsh123**
continuing what @ar37-rs started in #143 😇 
goal is
- [x] switch to arboard
- [x] move clipboard to be `app` level
- [x] support it on web

### CODE REVIEW

File: crates/notan_app/src/backend.rs [hunk 1]
```diff
@@ -25,6 +25,9 @@ pub trait Backend: Downcast {
     /// Returns the window implementation
     fn window(&mut self) -> &mut dyn WindowBackend;
 
+    #[cfg(feature = "clipboard")]
+    fn set_clipboard_text(&mut self, text: &str);
```
Comment (jawadsh123) [inline crates/notan_app/src/backend.rs:29]: question: should this go in `WindowBackend`?

File: crates/notan_winit/Cargo.toml [hunk 2]
```diff
@@ -18,12 +18,14 @@ notan_app = { path = "../notan_app", version = "0.7.1" }
 notan_glow = { path = "../notan_glow", version = "0.7.1" }
 notan_audio = { path = "../notan_audio", version = "0.7.1", optional = true }
 notan_oddio = { path = "../notan_oddio", version = "0.7.1", optional = true }
+notan_input = { path = "../notan_input", version = "0.7.1", optional = true }
```
Comment (jawadsh123) [inline crates/notan_winit/Cargo.toml:21]: this cuz we need modifier keys state to decide cut/copy/paste

File: crates/notan_app/src/backend.rs (hunk 1 above)
Comment (jawadsh123) [reply crates/notan_app/src/backend.rs:29]: I also wanted to add `get_clipboard_text`, but web's clipboard APIs are async and things got tricky real fast 🥲

**Review by jawadsh123 — COMMENTED** [review]
aight, ready for a review! (will add an example later)

**Comment by jawadsh123** [comment]
also...
although I like the general shape of it, there's one thing I'm unsure on and would love some feedback 🙏 
the `clipboard` feature gate as it stands is _all-or-nothing_, i.e. 
- when disabled, due conditional compilation, all of clipboard events/APIs are unavailable
- when enabled, they exist, and are backed by a system clipboard (arboard / web clipboard APIs)

there's also a slightly more slack approach, that egui takes
- when the feature is disabled, clipboard events/APIs still exist, but are backed by an emulated clipboard and not system clipboard
- when feature is enabled, these events/APIs are backed by system clipboard, and in case system clipboard is unavailable/fails, they fallback to emulated

after thinking a bunch I'm leaning towards egui's approach
though wanna know what others think before making a switch :)

**Comment by Nazariglez** [comment]
Outstanding work @jawadsh123 ! Thank you so much for it. I'll check it deeply soon, but right now it looks great to me. FWIW if you think that it's a good idea to follow the egui's design with the clipboard features, let's do it, it sounds good to me.

File: crates/notan_winit/Cargo.toml (hunk 2 above)
Comment (Nazariglez) [reply crates/notan_winit/Cargo.toml:21]: It makes sense to me.

File: crates/notan_app/src/backend.rs (hunk 1 above)
Comment (Nazariglez) [reply crates/notan_app/src/backend.rs:29]: I don't have a strong opinion about if it should be on `WindowBackend` or not. It makes sense to me to have it there, but at the same time, we have `open_link` on the backend, and both (`open_link` and `set_clipboard_text`) depend on external crates (not winit), so yeah, not strong opinion. For me, it's ok how you made it, we can always move it later if we think it doesn't fit well there. 

About the `get_clipboard_text` on browsers, I tried to block the UI thread until the promise is resolved, and well... no surprises, it doesn't work at all. 

I guess we can go now without it, it's an edge case, but eventually, we should think of a way to do it. Maybe return an `Option<String>`, it always returns `Some("")` on desktop as default (even if it's nothing on the clipboard) and on web we return `None` while the promise is resolved and then `Some("")` when it's resolved. Just an idea.

**Comment by jawadsh123** [comment]
this is ready!

the public clipboard APIs (events, funcs) now _exist_ even when the feature is disabled
reasoning goes, if a project disables clipboard for certain targets (maybe arboard doesn't support it, etc) the code should not break, just behave different (i.e. those events are never fired, `set_clipboard_text` has no effect)
this also sets us up to emulate an app local clipboard down the line if needed

**Review by Nazariglez — APPROVED** [review]
This is awesome! Thank you so much.

**Pull request merged**